src/Microsoft.Crank.AzureDevOpsWorker/Program.cs

```diff
@@ -160,57 +160,70 @@ private static async Task MessageHandler(ProcessMessageEventArgs args)
-                    while (driverJob.IsRunning)
+                    do
                     {
-                        if ((DateTime.UtcNow - driverJob.StartTimeUtc) > jobPayload.Timeout)
+                        if (retries > 0)
                         {
-                            throw new Exception($"{LogNow} Job timed out ({jobPayload.Timeout}). The timeout can be increased in the queued message.");
+                            Console.WriteLine($"{LogNow} Job failed, attempt ({retries + 1} out of {jobPayload.Retries + 1}).");
                         }
 
-                        var logs = driverJob.FlushStandardOutput().ToArray();
+                        // The DriverJob manages the application's lifetime and standard output
+                        driverJob = new Job("crank", arguments);
+
+                        driverJob.OnStandardOutput = log => Console.WriteLine(log);
+
+                        driverJob.Start();
 
-                        // Send any page of logs to the AzDo task log feed
-                        if (logs.Any())
+                        // Pump application standard output while it's running
+                        while (driverJob.IsRunning)
                         {
-                            var success = await devopsMessage.SendTaskLogFeedsAsync(String.Join("\r\n", logs));
+                            if ((DateTime.UtcNow - driverJob.StartTimeUtc) > jobPayload.Timeout)
+                            {
+                                throw new Exception($"{LogNow} Job timed out ({jobPayload.Timeout}). The timeout can be increased in the queued message.");
+                            }
 
-                            if (!success)
+                            var logs = driverJob.FlushStandardOutput().ToArray();
+
+                            // Send any page of logs to the AzDo task log feed
+                            if (logs.Any())
                             {
-                                Console.ForegroundColor = ConsoleColor.DarkYellow;
-                                Console.WriteLine($"{LogNow} SendTaskLogFeedsAsync failed. If the task was canceled, this jobs should be stopped.");
-                                Console.ResetColor();
+                                var success = await devopsMessage.SendTaskLogFeedsAsync(String.Join("\r\n", logs));
 
-                                driverJob.Stop();
-                            }                                
-                        }
+                                if (!success)
+                                {
+                                    Console.ForegroundColor = ConsoleColor.DarkYellow;
+                                    Console.WriteLine($"{LogNow} SendTaskLogFeedsAsync failed. If the task was canceled, this jobs should be stopped.");
+                                    Console.ResetColor();
+
+                                    driverJob.Stop();
```

Comment:
`continue;`?

Reply:
I see the point, so I removed this line and let the next part do it. But I used this in the timeout detection instead